test/parallel/test-shutil-rmtree.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+const shutil = require('shutil');
+
+const tmpdir = require('../common/tmpdir');
+
+const tmpDir = tmpdir.path;
+const baseDir = path.join(tmpDir, 'shutil-rmtree');
+
+function setupFiles() {
+  fs.mkdirSync(baseDir);
+  fs.mkdirSync(path.join(baseDir, 'dir1'));
+  fs.mkdirSync(path.join(baseDir, 'dir2'));
+  fs.mkdirSync(path.join(baseDir, 'dir2', 'dir3'));
+  fs.mkdirSync(path.join(baseDir, 'dir4'));
+
+  fs.closeSync(fs.openSync(path.join(baseDir, 'dir1', 'file1'), 'w'));
+  fs.closeSync(fs.openSync(path.join(baseDir, 'dir1', 'file2'), 'w'));
+  fs.closeSync(fs.openSync(path.join(baseDir, 'dir2', 'file3'), 'w'));
+  fs.closeSync(fs.openSync(path.join(baseDir, 'dir2', 'dir3', 'file4'), 'w'));
+  fs.closeSync(fs.openSync(path.join(baseDir, 'dir2', 'dir3', 'file5'), 'w'));
+}
+
+function assertGone(path) {
+  assert.throws(() => fs.readdirSync(path), {
+    code: 'ENOENT'
+  });
+}
+
+function test_rmtree_sync(path) {
+  setupFiles();
+
+  assert(fs.readdirSync(path));
+
+  shutil.rmtreeSync(path);
+  assertGone(path);
+}
+
+function test_invalid_path() {
+  assert.throws(() => shutil.rmtreeSync(1), {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+}
+
+function test_invalid_options(path) {
+  assert.throws(() => shutil.rmtreeSync(path, 1), {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+}
+
+function test_invalid_callback(path) {
+  assert.throws(() => shutil.rmtree(path, {}, 1), {
+    code: 'ERR_INVALID_CALLBACK'
+  });
+}
+
+function test_rmtree(path) {
+  setupFiles();
+
+  assert(fs.readdirSync(path));
+
+  shutil.rmtree(path, () => assertGone(path));
+}
+
```

Comment:
these tests are structured a bit differently than how we structure tests elsewhere in the codebase; which we usually put in blocks:

```js
// description of test.
{
  test.
}
```

not functions -- I don't hate this structure .. but might be good to make it consistent just to have one less thing to talk about.

Reply:
I was trying to model this after some of the existing `fs` tests and didn't notice that pattern. I'll take a look at some more existing tests. Any particularly good examples you can share?

Reply:
ignore me if you’re cribbing existing texts 👍

Reply:
Well, I'm trying. But I'd say I'm familiar with about 2% of the existing codebase. 😀